### PR TITLE
retry when abtest compare occurred an error

### PR DESCRIPTION
**Pull request by mahjonp**
Signed-off-by: mahjonp <junpeng.man@gmail.com>

### What problem does this PR solve? 

When startCheckConsistency occurred an error, it will fatal immediately, that't not what we expect.

### Check List 

Tests 

 - Unit test
 - E2E test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: pkg/pocket/core/check.go [hunk 1]
```diff
@@ -39,11 +40,24 @@ func (c *Core) startCheckConsistency() {
...
 		go func(t int) {
+			var (
+				result bool
+				err    error
+			)
 			log.Info("ready to compare data")
-			result, err := c.checkConsistency(false)
+
+			err = wait.PollImmediate(1*time.Minute, 10*time.Minute, func() (bool, error) {
+				result, err = c.checkConsistency(false)
```
Comment (you06) [inline pkg/pocket/core/check.go:50]: This error can be caused by database unavailable either data inconsistency, and retry is necessary is for the first condition only.

**Comment by you06** [comment]
Besides, the compare process will stop execution and waiting for sync first, move the error retry into this function would be better.
https://github.com/pingcap/tipocket/blob/8dfba30f71eebe6eb7aa77c38a6b8a4fc0447f5d/pkg/pocket/core/check.go#L249-L259

File: pkg/pocket/core/check.go (hunk 1 above)
Comment (mahjonp) [reply pkg/pocket/core/check.go:50]: Seems there will be a panic when data inconsistency occurred?

**Comment by mahjonp** [comment]
> Besides, the compare process will stop execution and waiting for sync first, move the error retry into this function would be better.
> https://github.com/pingcap/tipocket/blob/8dfba30f71eebe6eb7aa77c38a6b8a4fc0447f5d/pkg/pocket/core/check.go#L249-L259

Yes, that's better. I will move retrying here.

**Comment by mahjonp** [comment]
> Besides, the compare process will stop execution and waiting for sync first, move the error retry into this function would be better.
> https://github.com/pingcap/tipocket/blob/8dfba30f71eebe6eb7aa77c38a6b8a4fc0447f5d/pkg/pocket/core/check.go#L249-L259

The outer retry is required, it's used to catch the below error before compareData
https://github.com/pingcap/tipocket/blob/8dfba30f71eebe6eb7aa77c38a6b8a4fc0447f5d/pkg/pocket/core/check.go#L169-L172

File: pkg/pocket/core/check.go (hunk 1 above)
Comment (you06) [reply pkg/pocket/core/check.go:50]: Yes, so there should be only inconsistency and time exceeded errors.

Comment (mapleFU) [reply pkg/pocket/core/check.go:50]: Why not add a cause checking here?

**Pull request merged**